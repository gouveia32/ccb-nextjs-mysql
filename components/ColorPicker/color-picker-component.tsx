import React, { useState } from "react";
import { CirclePicker } from "react-color";
import PaletteIcon from "@material-ui/icons/Palette";
import { IconButton } from "@material-ui/core";
import { ColorPickerContent, ColorPickerMain } from "./color-picker.styles";

export interface ColorPickerProps {
  onChooseColor: (color: string) => void;
  edit?: boolean;
}

const ColorPicker: React.FC<ColorPickerProps> = ({
  onChooseColor,
  edit,
}: ColorPickerProps) => {
  const [pickerOpen, setPickerOpen] = useState(false);

  const renderPicker = (
    <ColorPickerContent edit={edit} onMouseLeave={() => onChooseColor("#fff")}>
      <CirclePicker
        onSwatchHover={(color) => onChooseColor(color.hex)}
        onChangeComplete={(color) => {
          onChooseColor(color.hex);
          setPickerOpen(false);
        }}
      />
    </ColorPickerContent>
  );

  return (
    <ColorPickerMain>
      {pickerOpen && renderPicker}
      <IconButton size={"small"} onClick={() => setPickerOpen(!pickerOpen)}>
        <PaletteIcon />
      </IconButton>
    </ColorPickerMain>
  );
};

export default ColorPicker;
