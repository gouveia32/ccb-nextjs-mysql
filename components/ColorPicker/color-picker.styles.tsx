import styled from "styled-components";
import { device } from "../../resources/styles/utils/media-query-utils";

export const ColorPickerMain = styled.div`
  position: relative;
`;

interface ColorPickerContentInterface {
  edit?: boolean;
}

export const ColorPickerContent = styled.div<ColorPickerContentInterface>`
  position: absolute;
  bottom: 3rem;
  ${(props) => (props.edit ? "right: 0" : "left: 0")};
  z-index: 3;
  padding: 1rem;
  border-radius: 1rem;
  background-color: white;
  border: 1px solid lightgray;

  @media only screen and ${device.mobileL} {
    right: 0;
    left: auto;
  }
`;
