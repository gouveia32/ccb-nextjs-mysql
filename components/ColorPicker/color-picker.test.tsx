import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import ColorPicker, { ColorPickerProps } from "./color-picker-component";

describe("ColorPicker component", () => {
  let wrapper: ShallowWrapper;
  let mockOnChooseColor: any;

  beforeEach(() => {
    mockOnChooseColor = jest.fn();

    const mockProps: ColorPickerProps = {
      edit: false,
      onChooseColor: mockOnChooseColor,
    };

    wrapper = shallow(<ColorPicker {...mockProps} />);
  });

  it("should render ColorPicker component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
