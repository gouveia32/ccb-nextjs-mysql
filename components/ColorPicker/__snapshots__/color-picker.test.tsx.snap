// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component should render ColorPicker component 1`] = `ShallowWrapper {}`;
